Migrate GUD documentation to DTMO. (#395)
mozilla/firefox-data-docs: src/tools/gud.md

# Mozilla Growth & Usage Dashboard
#### *overall: jmccrosky@ | data: jklukas@ | frontend: hulmer@ (updated 2020-1-15)*
___

**The Mozilla Growth & Usage Dashboard is in general release and can be found [here](https://growth-stage.bespoke.nonprod.dataops.mozgcp.net/).**

___

# Overview

The Mozilla Growth & Usage Dashboard (GUD) is a tool to visualize growth metrics in a standard way across Mozilla’s products.  It is intended to be quite intuitive to use, but we provide some brief documentation here.  Please reach out (`#gud` on slack) if anything is unclear or you are unsure how to use GUD for your purposes.

# Quick Start

> To visualize a particular metric (such as `MAU` or `Retention`) for a particular product:

Reviewer: Most of the terms in backticks in this article probably don't need them, or should be italicized with underscores instead. In particular, MAU and retention both seem like natural words that don't need any adornment.

Author: My intent was to call out words that correspond to UI elements in the dashboard, like the PRODUCT / USAGE CRITERIA selector, or specific menu items like metrics and such.  Do you think this is not helpful?  Or helpful but better done with italics or something?

Reviewer: I see your point. Let's roll with it.

1. By default, all metrics are shown.  To zoom into a particular metric, choose it from the `METRIC` selector on the left.

2. Choose the product of interest from the `PRODUCT / USAGE CRITERIA` selector on the left.  Some products may have multiple possible usage criteria, corresponding to different types of use of the product.

>     * New profile creation is considered one of these usage criteria; thus, to determine the rate of new profile creation for a product, select that product’s “`Profile Created`” usage criteria and look at the `DAU` metric.

Reviewer: ```suggestion
    * New profile creation is considered one of these usage criteria; thus, to determine the rate of new profile creation for a product, select that product’s "`Profile Created`" usage criterion and look at the `DAU` metric.
```

You may also restrict to particular slices (for example, country) and to a particular date range using the additional selectors on the left.

> # Data Model

Reviewer: This explanation largely overlaps with existing docs for the Exact MAU tables: https://docs.telemetry.mozilla.org/datasets/bigquery/exact_mau/reference.html

We should probably consolidate so that we don't duplicate these explanations, and make sure to reference whatever the canonical location is from the other location.

Author: I see your point, but also kind of feel it really needs to be in both places.  I think it's pretty essential to really understanding how to use GUD, so want to have it here, but also see that it needs to be in the exact MAU table reference.  Does markdown support an include?  Maybe that's a partial solution to at least restrict that they're in sync...  I'm also open to being convinced that one could just link to the other, but don't see it that way now.

Reviewer: I agree that it's pretty central in both cases. I'm okay with duplicating for now, but perhaps we should at least make sure we point out in both articles pretty early on that a very similar data model exists for the other case, so these are obviously cross-linked. I don't want to block this on figuring out a technical solution.

Author: Sounds fair.  I made the change to better cross-link.  I'll wait until Friday morning to merge in case you want to review that, or any of the other reviewers want to weigh in.

## Usage Criteria

### Firefox (Desktop)

#### Any Firefox Activity

The profile has sent a telemetry ping from Firefox Desktop on the day in question.

#### New Firefox Desktop Profile Created

The count of the number of new profiles created on the day in question.  Note this uses a main ping methodology for support for our entire history, rather than the newer profile_creation_ping.

#### Visited 5 URIs

The profile sent a main ping on the day in question that indicates that the profile opened at least 5 URIs.

#### Opened DevTools

The profile sent a main ping on the day in question that indicates that the developer tools feature was activated at least once.
Fenix

#### Any Fenix Activity

The profile has sent a telemetry ping from Fenix on the day in question.
Firefox Accounts

#### Any Firefox Account Activity

The profile has sent a telemetry ping from Firefox Account on the day in question.  This corresponds to “active” events in Amplitude.

### Fennec

#### Any Fennec Android Activity

The profile has sent a telemetry ping from Fennec Android on the day in question.

#### Any Fennec iOS Activity

The profile has sent a telemetry ping from Fennec iOS on the day in question.

> ### Focus

Reviewer: ```suggestion

### Focus
```

The profile has sent a telemetry ping from FirefoxConnect on the day in question.

## Dimensions

During a period of time, a particular profile may take on more than one value within a dimension.  For example, a profile may use release channel and beta channel browsers in a single day, resulting in two pings for the day with different values of channel.  Similarly, a profile may be used in one country one day and another country the next day, resulting in two pings for two consecutive days that have different values of country.  Generally, for a metric, we need to assign a profile to one particular value for each dimension.  As such we define the value for each dimension within a day by taking the most frequent value seen in that day, breaking ties by taking the value that occurs last.  We then use the values from the last day within the data that the metric examines.  This is documented in more detail [here](https://docs.telemetry.mozilla.org/datasets/bigquery/exact_mau/reference.html#slice).

### OS

Whether the profile’s pings were sent from Mac OS, Windows, or Linux.

### Language

The language (as extracted from the locale) of the profile’s pings.

### Country

> The country (as extracted through IPGeo) of the profile’s pings.

Reviewer: ```suggestion
The country (as extracted through GeoIP lookup) of the profile’s pings.
```

### Channel

The channel of the profile’s pings.

# Bug reports / Feature Requests

> Please direct all feedback to ``#gud` on slack.

Reviewer: ```suggestion
Please direct all feedback to `#gud` on slack.
```
